src/lifecycle_point_stamped_incr.cpp

```diff
@@ -0,0 +1,955 @@
+///  `~/z_offset (float) [default 0.0]`             - The offset for the message's zero value
+///
+///  `~/always_enable (bool) [default false]`      - Whether control input is always enabled (USE WITH CAUTION)
+///
+///  `~/input_device_config_file (std::string) [default "dualshock4_mapping"]`      - Chosen input device config file
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+
+#include <string>
+#include <cmath>
+
+#include "yaml-cpp/yaml.h"
+#include <iostream>
+#include <ament_index_cpp/get_package_share_directory.hpp>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+// Additional libraries for Lifecycle nodes
+#include <thread>
+#include <utility>
+#include "lifecycle_msgs/msg/transition.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/publisher.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
+#include "rcutils/logging_macros.h"
+#include "std_msgs/msg/string.hpp"
+
+using std::string;
+
+static geometry_msgs::msg::PointStamped command;
+static sensor_msgs::msg::Joy latest_joy_state;
+static bool fresh_joy_state;
+static bool always_enable = false;
+static const float rate_of_change_denom = 1 * std::pow(10, 5); // USER CAN ADJUST, KEEP IT EXTREMELY SMALL
+
+float curr_x_max = 1.0;
+float curr_y_max = 1.0;
+float curr_z_max = 1.0;
+static float x_max;
+static float y_max;
+static float z_max;
+static float alt_x_max;
+static float alt_y_max;
+static float alt_z_max;
+static float boundary_radius;
+float lin_rate_chg_fac;
+static float x_offset;
+static float y_offset;
+static float z_offset;
+static bool x_flip;
+static bool y_flip;
+static bool z_flip;
+
+/// @brief The input type of an input (Axis, Trigger, Button, None)
+enum class InputType
+{
+    Axis,
+    Trigger,
+    Button,
+    None
+};
+
+static const int UNUSED_INDEX = -1;
+static const InputType UNUSED_TYPE = InputType::None;
+
+/// @brief An object representing a particular function's (e.g. move forward, move left) input,
+///        containing said input's index in the received joy message and that input's input type
+class MovementInput
+{
+    public:
+        /// @brief Index number that correlates with its position in the joy message array
+        int index;
+        /// @brief InputType that indicates the type of input it is (Axis, Trigger, Button, None)
+        InputType type;
+
+        /// @brief Default constructor. Initializes index to UNUSED_INDEX and type to InputType::None
+        MovementInput() : index(UNUSED_INDEX), type(InputType::None) {}
+        /// @brief Constructor with provided index and type parameters.
+        /// @param index_no The index of the joy message array.
+        /// @param input_type The type of input.
+        MovementInput(int index_no, InputType input_type) : index(index_no), type(input_type) {}
+};
+
+// /// Additional parameter helper functions developed my NU MSR
```

Comment:
No commented-out code

Reply:
Left a few commented out debugging log statements, but the rest is all gone